examples/Microsoft.Spark.CSharp.Examples/Sql/VectorUdfs.cs

```diff
@@ -50,45 +53,28 @@ public void Run(string[] args)
                         new StructField("age", new IntegerType()),
                         new StructField("nameCharCount", new IntegerType())
                     }),
-                    r => CountCharacters(r, "age", "name"))
+                    r => CountCharacters(r, "name"))
                 .Show();
 
             spark.Stop();
         }
 
-        private static RecordBatch CountCharacters(
-            RecordBatch records,
-            string groupFieldName,
+        private static FxDataFrame CountCharacters(
+            FxDataFrame dataFrame,
             string stringFieldName)
         {
-            int stringFieldIndex = records.Schema.GetFieldIndex(stringFieldName);
-            StringArray stringValues = records.Column(stringFieldIndex) as StringArray;
-
             int characterCount = 0;
 
-            for (int i = 0; i < stringValues.Length; ++i)
+            PrimitiveColumn<int> characterCountColumn = new PrimitiveColumn<int>(stringFieldName + "CharCount", dataFrame.RowCount);
+            for (long i = 0; i < dataFrame.RowCount; i++)
             {
-                string current = stringValues.GetString(i);
-                characterCount += current.Length;
+                characterCount += ((string)dataFrame[stringFieldName][i]).Length;
+                characterCountColumn[i] = characterCount;
             }
 
-            int groupFieldIndex = records.Schema.GetFieldIndex(groupFieldName);
-            Field groupField = records.Schema.GetFieldByIndex(groupFieldIndex);
-
-            // Return 1 record, if we were given any. 0, otherwise.
```

Comment:
This isn't doing what it was originally doing.

Originally it would only return 1 record with 2 columns: "Age" and "name_CharCount".

However, you are now return N records, where N is the number of records passed in to the UDF.